frontend/src/vault/keygen/state/currentServerUrl.ts

```diff
@@ -0,0 +1,6 @@
+import { getValueProviderSetup } from '../../../lib/ui/state/getValueProviderSetup';
+
+export const {
```

Comment:
Good point. The idea here is that a piece of state may be accessible just through a hook, yet in some situation it would require a provider.

E.g. It could export only `useCurrentServerUrl` or `useCurrentServerUrl` + `CurrentServerUrlProvider`.